src/Symfony/Component/ObjectMapper/ObjectMapper.php

```diff
@@ -0,0 +1,326 @@
+            }
+
+            if (!$mappings && $targetRefl->hasProperty($propertyName)) {
+                $value = $this->getSourceValue($source, $mappedTarget, $this->getRawValue($source, $propertyName), $objectMap);
+                $this->storeValue($propertyName, $mapToProperties, $ctorArguments, $value);
+            }
+        }
+
+        if (!$mappingToObject && $ctorArguments) {
+            try {
+                $constructor?->invokeArgs($mappedTarget, $ctorArguments);
+            } catch (\ReflectionException $e) {
+                throw new MappingException($e->getMessage(), $e->getCode(), $e);
+            }
+        }
+
+        foreach ($mapToProperties as $property => $value) {
+            $this->propertyAccessor ? $this->propertyAccessor->setValue($mappedTarget, $property, $value) : ($mappedTarget->{$property} = $value);
+        }
+
+        if ($objectMapInitialized) {
+            $objectMap = null;
+        }
+
+        return $mappedTarget;
+    }
+
+    private function getRawValue(object $source, string $propertyName): mixed
+    {
+        return $this->propertyAccessor ? $this->propertyAccessor->getValue($source, $propertyName) : $source->{$propertyName};
+    }
+
+    private function getSourceValue(object $source, object $target, mixed $value, \SplObjectStorage $objectMap, ?Mapping $mapping = null): mixed
+    {
+        if ($mapping?->transform) {
+            $value = $this->applyTransforms($mapping, $value, $source);
+        }
+
+        if (
+            \is_object($value)
+            && ($innerMetadata = $this->metadataFactory->create($value))
+            && ($mapTo = $this->getMapTarget($innerMetadata, $value, $source))
+            && (\is_string($mapTo->target) && class_exists($mapTo->target))
+        ) {
+            $value = $this->applyTransforms($mapTo, $value, $source);
+
+            if ($value === $source) {
+                $value = $target;
+            } elseif ($objectMap->contains($value)) {
+                $value = $objectMap[$value];
+            } else {
+                $value = $this->map($value, $mapTo->target);
+            }
+        }
+
+        return $value;
+    }
+
+    /**
+     * Store the value either the constructor arguments or as a property to be mapped.
+     *
+     * @param array<string, mixed> $mapToProperties
+     * @param array<string, mixed> $ctorArguments
+     */
+    private function storeValue(string $propertyName, array &$mapToProperties, array &$ctorArguments, mixed $value): void
+    {
+        if (\array_key_exists($propertyName, $ctorArguments)) {
+            $ctorArguments[$propertyName] = $value;
+
+            return;
+        }
+
+        $mapToProperties[$propertyName] = $value;
+    }
+
+    /**
+     * @param callable(): mixed $fn
+     */
+    private function call(callable $fn, mixed $value, object $object): mixed
+    {
+        try {
+            $refl = new \ReflectionFunction(\Closure::fromCallable($fn));
```

Comment:
instead of using reflection + name-based convention on the "object" argument
let's never pass the object when the callable is a string, to help using native functions